ibc-apps/ics20-transfer/types/src/coin.rs

```diff
@@ -76,7 +76,10 @@ where
                     .chars()
                     .all(|x| x.is_alphanumeric() || VALID_DENOM_CHARACTERS.contains(x))
             })
-            .ok_or_else(|| TokenTransferError::InvalidCoin(coin_str.to_string()))?;
+            .ok_or_else(|| TokenTransferError::FailedToParseType {
```

Comment:
Following our chats, have you decided to stick with `FailedToParseType`, or should it be changed back to `InvalidCoin`?
If we’re going with `FailedToParseType`, the `expected` field is misleading IMO, as it's not representing the actual expected value.

Reply:
What do you think would make more sense semantically for this variant?

Reply:
I'd rather having them back to `InvalidCoin `, `InvalidTrace` as it was before.

I am suggesting as based on the pattern we’ve used for `expected` and `actual` fields in other cases, they represent **values** rather than **types**. Here the `expected` field is showing a type (like "Coin" or "TracePath"), but the `actual` field contains a value. They are not comparable and inconsistent with how we’ve defined similar variants elsewhere.

Reply:
[2952291](https://github.com/cosmos/ibc-rs/pull/1325/commits/29522914999b02c31f3e933d7cfc273f2c24a916)